### PR TITLE
#2. This Pointer & #4. Arrays of objects

### PR DESCRIPTION
**Pull request by rajuTK9**
The PR explains the following topics:
- This Pointer
- Arrays of Objects

In this PR, I have created a few more examples to explain the feature.
In the next PR, I will be working on the core feature of my project.

Check out the explanation video [here](https://youtu.be/iHsgU6CDaFU).

### CODE REVIEW

File: src/main/java/Expense.java [hunk 1]
```diff
@@ -1,17 +1,91 @@
...
 
-    public Expense(int expenseId, String description, double amount, User paidBy) {
+    public Expense(int expenseId, String description, double amount, User paidBy, Group group) {
         this.expenseId = expenseId;
         this.description = description;
         this.amount = amount;
         this.paidBy = paidBy;
+        this.group = group;
+        this.benefactors = new ArrayList<>();
+    }
```
Comment (th3c0d3br34ker) [inline src/main/java/Expense.java:24]: `individualShare` wasnt initialized here.

File: src/main/java/Expense.java [hunk 2]
```diff
@@ -1,17 +1,91 @@
...
+    public List<User> getBenefactors() {
+        return benefactors;
+    }
+
+    public void addBenefactor(User user) {
+        benefactors.add(user);
+    }
+
+    public double getIndividualShare() {
+        return individualShare = calculateIndividualShare();
```
Comment (th3c0d3br34ker) [inline src/main/java/Expense.java]: this looks like a redundant assignment,

doesn't the `calculateIndividualShare`  act as a getter function for individual share? please callout the logic this code is structured this way.

am i missing some functionality that you plan to implement in future?

File: src/main/java/Group.java [hunk 3]
```diff
@@ -16,13 +17,43 @@ public Group(int groupId, String name) {
...
+    }
+
     public void addExpense(Expense expense) {
         expenses.add(expense);
     }
 
-    // Getters for attributes
+    // toString method to print group details
+    @Override
+    public String toString() {
```
Comment (th3c0d3br34ker) [inline src/main/java/Group.java:55]: good addition, makes it simple to print the expenses in all places.

**Review by th3c0d3br34ker — COMMENTED** [review]
hey @rajuTK9, thanks for the timely submission.

please resolve the comments.

also, please schedule a viva with me [here](https://calendar.kalvium.com/jainam-desai/30min).

File: src/main/java/Expense.java (hunk 1 above)
Comment (rajuTK9) [reply src/main/java/Expense.java:24]: Thanks for pointing it our. I had missed this and I have resolved it and pushed again. Kindly check it out.

File: src/main/java/Expense.java (hunk 2 above)
Comment (rajuTK9) [reply src/main/java/Expense.java]: You are correct, Jainam. This is a redundant assignment.

I have resolved this in the new commit. Please check it out and let me know if still any mistakes are there. 
Thank you!

**Review by th3c0d3br34ker — APPROVED** [review]
approved during the viva

**Pull request merged**